drivers/bridge/bridge.go

```diff
@@ -984,10 +985,12 @@ func (d *driver) CreateEndpoint(nid, eid string, ifInfo driverapi.InterfaceInfo,
 		}
 	}
 
-	// Program any required port mapping and store them in the endpoint
-	endpoint.portMapping, err = n.allocatePorts(epConfig, endpoint, config.DefaultBindingIP, d.config.EnableUserlandProxy)
-	if err != nil {
-		return err
+	if !config.Internal {
```

Comment:
This will not be needed when https://github.com/docker/libnetwork/pull/810 goes in.
Libnetwork will not send the network driver the port mapping information if the network is internal.

But for now it is fine, given this PR will likely be merged first.